### DISCUSSION TITLE
Obtain key parameters in ElastoDyn from WindIO yaml file.

### DISCUSSION THREAD

**Discussion by yangpy88** [IEAWindTask37/IEA-22-280-RWT#104] [Q&A] [answered]
Hello guys,

I'm trying to build connections between the parameters defining in OpenFAST's ElastoDyn and the materials of wind turbine configuration stored in WindIO yaml file, but have some confusions on the nacelle system. Let's just focus on this 22MW wind turbine. How do we get `GenIner` and `NacYIner`? It's not so straightforward as one can directly see `YawBrMass` is equal to `components.naccelle.drivetrain.elastic_properties_mb.yaw_mass` in the WindIO yaml file. I guess there are some mathematical approaches? Additionally, `NacMass` of 821239.8004933242 is not exactly equal to `components.naccelle.drivetrain.elastic_properties_mb.system_mass` of 821009.0477187126. What causes the slight difference? All files discussed here are referred to version 1.0.1, i.e., the newest released version. Thank you! 

Kind regards,

**Comment by ptrbortolotti** [comment, marked as answer]
Hello,
`drivese.generator_rotor_I` is used for `GenIner`, see [here](https://github.com/WISDEM/WEIS/blob/faf0efcc6a57e81bf13aeae603c1d002836bd5ad/weis/glue_code/glue_code.py#L457C66-L457C85) and [here](https://github.com/WISDEM/WEIS/blob/faf0efcc6a57e81bf13aeae603c1d002836bd5ad/weis/aeroelasticse/openmdao_openfast.py#L932)

`NacYIner` is computed according to this formula [here](https://github.com/WISDEM/WEIS/blob/faf0efcc6a57e81bf13aeae603c1d002836bd5ad/weis/aeroelasticse/openmdao_openfast.py#L925)

The tiny difference in nacelle mass is likely just coming from different iterations

**Reply by yangpy88** [reply]
Dear  Pietro,

Thanks for your answers. However, I cannot find definitions of the tower inertia in the yaml file. But `tower_I_base` is initialized as an array with all zeros if I understand the code correctly. Are they the intermediate variables during the computation with WEIS? How can I get the value of tower yaw inertia if I only have the yaml file? Thanks.

Kind regards,

**Reply by ptrbortolotti** [reply]
hello, unfortunately the yaml alone for WindIO v1.0 is not sufficient to generate a full OpenFAST model. Note that we are working to close this gap, and the target is to release WindIO v2.0 by the end of March 2025

**Reply by yangpy88** [reply]
OK, I will be looking forward to the new version. Thank you.
